plugins/languages/treesitter/treesitter.nix

```
@@ -6,210 +6,303 @@
-          keymaps = {
-            initSelection = keymap "gnn";
-            nodeIncremental = keymap "grn";
-            scopeIncremental = keymap "grc";
-            nodeDecremental = keymap "grm";
-          };
+          init_selection = mkKeymap "gnn";
+          node_incremental = mkKeymap "grn";
+          scope_incremental = mkKeymap "grc";
+          node_decremental = mkKeymap "grm";
         };
+    };
 
-      indent = mkEnableOption "tree-sitter based indentation";
+    indent = helpers.defaultNullOpts.mkBool true ''
+      Whether to enable tree-sitter based indentation.
+    '';
 
-      folding = mkEnableOption "tree-sitter based folding";
+    ensure_installed = helpers.defaultNullOpts.mkListOf types.str [ "all" ] ''
+      Either "all" or a list of languages to ensure installing.
+    '';
 
-      languageRegister = mkOption {
-        type = with types; attrsOf (either str (listOf str));
-        description = ''
-          This is a wrapping of the `vim.treesitter.language.register` function.
-          Register specific parsers to one or several filetypes.
-          The keys are the parser names and the values are either one or several filetypes.
-        '';
-        default = { };
-        example = {
-          cpp = "onelab";
-          python = [
-            "myFiletype"
-            "anotherFiletype"
-          ];
-        };
-      };
+    ignore_install = helpers.defaultNullOpts.mkListOf types.str [ ] ''
+      List of parsers to ignore installing (for "all")
+    '';
 
-      grammarPackages = mkOption {
-        type = with types; listOf package;
-        default = cfg.package.passthru.allGrammars;
-        description = "Grammar packages to install";
+    parser_install_dir = helpers.defaultNullOpts.mkStr "$XDG_DATA_HOME/nvim/treesitter" ''
+      Location of the parsers to be installed by the plugin (only needed when nixGrammars is disabled).
+      This default might not work on your own install, please make sure that $XDG_DATA_HOME is set if you want to use the default. Otherwise, change it to something that will work for you!
+    '';
+  };
+
+  settingsExample = {
+    additional_vim_regex_highlighting = true;
+    ensure_installed = true;
+    ignore_install = [ "rust" ];
+    indent = true;
+    parser_install_dir = "$XDG_DATA_HOME/nvim/treesitter";
+
+    highlight = {
+      enable = true;
+      disable = [ "rust" ];
+      custom_captures = { };
+    };
+
+    incremental_selection = {
+      enable = true;
+
+      keymaps = {
+        init_selection = "gnn";
+        node_incremental = "grn";
+        scope_incremental = "grc";
+        node_decremental = "grm";
       };
+    };
+  };
+
+  extraOptions = {
+    folding = mkOption {
+      type = types.bool;
+      default = true;
+      example = true;
+      description = "Whether to enable tree-sitter based folding.";
+    };
+
+    gccPackage = helpers.mkPackageOption {
+      name = "gcc";
+      default = pkgs.gcc;
+      example = pkgs.gcc;
```

The example will need to use `literalExpression` otherwise it will render as `<derivation gcc>` or similar.

Same applies to the default:

```suggestion
      defaultText = literalExpression "pkgs.gcc";
      example = literalExpression "pkgs.gcc";
```

---

In this specific case, I'm not sure the example is useful, since it is the same as the default.

Either remove the example, or set it to something non-default.